lib/hmmlearn/base.py

```diff
@@ -596,6 +678,21 @@ def _compute_log_likelihood(self, X):
             model states, i.e., ``log(p(X|state))``.
         """
 
+    def _compute_likelihood(self, X):
+        """Computes per-component log probability under the model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Feature matrix of individual samples.
+
+        Returns
+        -------
+        logprob : array, shape (n_samples, n_components)
+            Log probability of each sample in ``X`` for each of the
+            model states.
+        """
```

Comment:
Minor nit: this could go above _compute_log_likelihood (likewise in the subclasses), so that the methods related to the scaling version always comes first.

Also, you could do add an implementation like
```python
if self._compute_log_likelihood != _BaseHMM._compute_log_likelihood.__get__(self):  # prevent recursion
    return np.exp(self._compute_log_likelihood(X))
else:
    raise NotImplementedError("Must be overridden in subclass")
```
and likewise to _compute_log_likelihood; this way, subclasses only need to implement *one* of _compute_likelihood and _compute_log_likelihood, whichever is more practical (the overhead from the extra function call should be negligible).

This should also be mentioned at the very end of tutorial.rst, where overridable methods are listed.

Reply:
Fair suggestion.  I was planning to prepare a second MR to implement the non-log likelihood methods.  I can do this for now.